### PR TITLE
[Core][MappingApplication] Make MapperFactory use Registry

### PR DESCRIPTION
**Pull request by jcotela**
**📝 Description**
Refactoring the MapperFactory in order to use the Registry infrastructure instead of its internal registered_mappers list.

The motivation for the change is related to the new gtests and the need to load application dependencies dynamically. Consider a test suite that loads MappingApplication as a runtime dependency (because the tested application does not have compile-time dependencies to it and does not link to it)

1. Kratos Core imported (including MapperFactory)
2. MappingApplication imported dynamically on SetUpTestSuite
3. For each test:
 3.1 Mappers registered in MapperFactory on each test start (when the test suite imports its dependencies in the kernel)
 3.2 Test run
 3.3 Applications de-registered
4. MappingApplication unloaded on TearDownTestSuite
5. MapperFactory mapper list deallocated, but the mapper class definitions are no longer available -> Memory error

If the MapperFactory uses the registry instead, mappers are automatically registered when MapperApplication is de-registered in 3.3, so the deallocation in 5 is no longer a problem.

**🆕 Changelog**
- MapperFactory uses Registry to hold registered mappers instead of an internal static variable.

### CODE REVIEW

**Review by loumalouomega — COMMENTED** [review]
Okay from my side, maybe we should wait for @philbucher  approval

File: kratos/factories/mapper_factory.h
Comment (philbucher) [inline kratos/factories/mapper_factory.h:170]: those names are only internally used in the registry right? I.e. nothing the user has to deal with?

**Review by philbucher — DISMISSED** [review]

File: kratos/factories/mapper_factory.h
Comment (jcotela) [reply kratos/factories/mapper_factory.h:170]: The users of MapperFactory should be able to continue interacting with it as usual. One could get the mappers directly form the registry using those paths, but this is not used at the moment.

**Review by philbucher — APPROVED** [review]

**Pull request merged**
